.github/workflows/wheel.yml

```
@@ -2,88 +2,125 @@ name: Build binary wheel
+              && cibuildwheel --print-build-identifiers --platform macos --arch arm64 | grep cp |  jq -nRc '{"only": inputs, "os": "macos-14"}' \
+              && cibuildwheel --print-build-identifiers --platform windows --arch AMD64 | grep cp |  jq -nRc '{"only": inputs, "os": "windows-latest"}'
+            } | jq -sc
+          )
+          echo "include=$MATRIX_INCLUDE" >> $GITHUB_OUTPUT
+          cat $GITHUB_OUTPUT
+        working-directory: packages/vaex-core/
+
   build_wheels:
-    name: Build wheel on ${{ matrix.os }}
+    needs: build_wheels_matrix
     runs-on: ${{ matrix.os }}
+    name: Build ${{ matrix.only }}
+
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, macOS-11, windows-latest]
-        python-version: [36, 37, 38, 39, 310]
+        include: ${{ fromJson(needs.build_wheels_matrix.outputs.include) }}
+
+    permissions:
+      contents: write # for uploading release assets
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v4
       with:
         submodules: true
 
     - name: Copy dll
-      if: (matrix.os == 'windows-latest')
+      if: matrix.os == 'windows-latest'
       uses: ./ci/actions/windll
 
-    - uses: actions/setup-python@v2
-      name: Install Python
-      with:
-        python-version: '3.7'
-
-    - name: Install cibuildwheel
-      run: |
-        python -m pip install cibuildwheel==2.8.1
-
-    - name: Install twine/wheel
-      run: |
-        python -m pip install twine wheel
-
     - name: chores
       if: matrix.os != 'windows-latest'
       run: |
-        mkdir dist
         mkdir packages/vaex-core/bin
         cp bin/install_pcre.sh packages/vaex-core/bin/
 
-    - name: Build vaex-core (windows)
-      if: (matrix.os == 'windows-latest')
-      env:
-        CIBW_BUILD: cp${{ matrix.python-version }}*64
-        CIBW_ENVIRONMENT_LINUX: "CFLAGS='-Wl,-strip-all' CXXFLAGS='-Wl,-strip-all'"
-      run: |
-        python -m cibuildwheel --output-dir dist packages/vaex-core
+    - name: Set up QEMU
+      if: runner.os == 'Linux'
+      uses: docker/setup-qemu-action@v3
 
-    - name: Build vaex-core (linux, osx)
-      if: (matrix.os != 'windows-latest')
+    - uses: pypa/cibuildwheel@v2.17.0 # sync version with pip install cibuildwheel above
+      with:
+        only: ${{ matrix.only }}
+        package-dir: packages/vaex-core/
+        output-dir: packages/vaex-core/dist/
       env:
-        CIBW_BEFORE_BUILD: bin/install_pcre.sh
-        CIBW_BUILD: "cp${{ matrix.python-version }}*64 cp${{ matrix.python-version }}-macosx-*"
-        CIBW_ENVIRONMENT_LINUX: "CFLAGS='-Wl,-strip-all' CXXFLAGS='-Wl,-strip-all'"
+        CIBW_BEFORE_BUILD: ${{ startswith(matrix.os, 'ubuntu') && 'bash bin/install_pcre.sh' || startswith(matrix.os, 'macos') && 'sudo -E bash bin/install_pcre.sh' || '' }}
         CIBW_BUILD_VERBOSITY: 2
-        # needed for m1 builds
+        CIBW_MANYLINUX_X86_64_IMAGE: manylinux_2_28
```

```suggestion
        CIBW_TEST_COMMAND: python -c "import vaex; vaex.example()"
        CIBW_MANYLINUX_X86_64_IMAGE: manylinux_2_28
```

maybe this won't work because cibuildwheel only installed vaex-core (and it's dependencies) at this point

---

That is true, we should not use hdf5 for this. 
```
python -c "import vaex; print(vaex.from_arrays(x=[1,2]))"                                          
```
This should be ok